File: cohesiv/models/tag.go
package models

import (
	"time"

	pb "github.com/SingularityInteractive/cohesiv/cohesiv"
)

// TagModel as represented in datastore
type TagModel struct {
	Name      string          `gorm:"primary_key"`
	Resources []ResourceModel `gorm:"many2many:tag_resources;"`
	CreatedAt time.Time
	UpdatedAt time.Time
}

// TableName renames the table on migration with gorm
func (TagModel) TableName() string {
	return "tags"
}

// ToProto converts a db tag to grpc tag
func (t *TagModel) ToProto() *pb.Tag {
	return &pb.Tag{
		Name: t.Name,
	}
}

File: cohesiv/models/tag.ts
import * as Sequelize from 'sequelize'
import sequelize from '../db'
import Resource from './resource'

const Tag = sequelize.define(
  'Tag',
  {
    name: { type: Sequelize.STRING, primaryKey: true, field: 'name' }
  },
  {
    tableName: 'tags',
    underscored: true
  }
)

Tag.belongsToMany(Resource, { as: 'Resources', through: 'tag_resources', foreignKey: 'tag_name' })

Tag.sync({
  force: process.env.NODE_ENV === 'production' ? false : true
})

export default Tag
